### PR TITLE
Add To Cart Block

### PR DESCRIPTION
**Pull request by HardeepAsrani**
A simple WooCommerce Add to Cart block. Fixes: https://github.com/Codeinwp/gutenberg-blocks/issues/734

Can you review this to see if it meets the requirement? @mghenciu

### CODE REVIEW

**Comment by mghenciu** [comment]
Sure! Please send me the build file @HardeepAsrani , when you can.
I tried to build the files myself, but failed yet again.

**Comment by HardeepAsrani** [comment]
@mghenciu Sent you on Slack.

**Comment by mghenciu** [comment]
Looks good @HardeepAsrani .
I like the idea with Visibility Conditions, should be quite useful.

A few things would be nice to have:
- alignment options for the button, now it stays aligned to left all the time. 
This options would be helpful for cases like this: 
image
or this: 
image
- at this point the button background color is different from the Primary button, and from what I remember it should use Primary button styling:
Screenshot 2021-08-16 at 11 28 40


fyi: I tested only design related things, not full functionality.

**Comment by HardeepAsrani** [comment]
@mghenciu 

"at this point the button background color is different from the Primary button, and from what I remember it should use Primary button styling:"

There's no Primary Button style when it comes to Button Blocks. There's only one default style, rest are however themes define them. Do you think we should add color options to it?

**Comment by mghenciu** [comment]
> @mghenciu
> 
> "at this point the button background color is different from the Primary button, and from what I remember it should use Primary button styling:"
> 
> There's no Primary Button style when it comes to Button Blocks. There's only one default style, rest are however themes define them. Do you think we should add color options to it?

Thinking forward, yes, probably it would make sense to have color options, as we discussed before that the blocks should work as expected with other themes.
For our case I was thinking about something like this controls, which makes the styling process very easy/eficient:
Screenshot 2021-08-17 at 09 40 48
you just add a button and select the style inherited from Customizer. I know the buttons from the screenshot above have color controls, but I am referring more to the fact that with 1 click, users have a consistent button.

_Long story short, the user story would sound something like this:_
As **a shop owner**, I want to **easily customize the Add to Cart block colors**, so it's **consistent with my overall Shop style** & I **don't have to use any Custom css** to modify the buttons.  

cc. @JohnPixle

**Comment by HardeepAsrani** [comment]
Or another, bit crazier, way is to just use Otter's Button block for this. That way it will have all the options, color/icon and such, but it will also be a far more complicated block (too many choices for users).

So yea, which poison do we need? Simple or everything? @mghenciu

**Comment by JohnPixle** [comment]
I vote for a simple solution here, without any dependencies on custom CSS or an Otter component.

In the best case scenario, user is using Neve, and he can have consistent buttons (automatically apply the primary or the secondary button if the theme supports these,

In worst case, for users who don't use Neve, we simply give him the default  button style, as defined in their theme.

Let me know your thoughts.

**Comment by HardeepAsrani** [comment]
@JohnPixle We went that way. Thanks!
